build_tools/packaging/linux/build_package.py

```diff
@@ -685,99 +799,111 @@ def download_and_extract_artifacts(run_id, gfxarch):
         sys.exit(1)
 
 
-def clean_artifacts_dir(clean_all):
-    """Clean the artifacts directory
-
-    Parameters:
-    clean_all : Flag to clean artifacts download directory
+def clean_rpm_build_dir():
+    """Clean the rpm build directory
 
+    Parameters: None
     Returns: None
     """
-    if clean_all:
-        if os.path.exists(ARTIFACTS_DIR) and os.path.isdir(ARTIFACTS_DIR):
-            shutil.rmtree(ARTIFACTS_DIR)
-            print(f"Removed directory: {ARTIFACTS_DIR}")
+    if os.path.exists(RPM_CONTENTS_DIR) and os.path.isdir(RPM_CONTENTS_DIR):
+        shutil.rmtree(RPM_CONTENTS_DIR)
+        print(f"Removed directory: {RPM_CONTENTS_DIR}")
+
 
+def clean_debian_build_dir():
+    """Clean the debian build directory
+
+    Parameters: None
+    Returns: None
+    """
     if os.path.exists(DEBIAN_CONTENTS_DIR) and os.path.isdir(DEBIAN_CONTENTS_DIR):
         shutil.rmtree(DEBIAN_CONTENTS_DIR)
         print(f"Removed directory: {DEBIAN_CONTENTS_DIR}")
-    if os.path.exists(RPM_CONTENTS_DIR) and os.path.isdir(RPM_CONTENTS_DIR):
-        shutil.rmtree(RPM_CONTENTS_DIR)
-        print(f"Removed directory: {RPM_CONTENTS_DIR}")
+
+
+def clean_package_build_dir(artifacts_dir):
+    """Clean the package build directories
+
+    If artifactory directory is provided, clean the same as well
+
+    Parameters:
+    artifacts_dir : Directory where artifacts are stored
+
+    Returns: None
+    """
+    print_function_name()
+
+    clean_rpm_build_dir()
+    clean_debian_build_dir()
 
     PYCACHE_DIR = "__pycache__"
     if os.path.exists(PYCACHE_DIR) and os.path.isdir(PYCACHE_DIR):
         shutil.rmtree(PYCACHE_DIR)
         print(f"Removed directory: {PYCACHE_DIR}")
 
+    if os.path.exists(artifacts_dir) and os.path.isdir(artifacts_dir):
+        shutil.rmtree(artifacts_dir)
```

Comment:
(comment applies throughout) Can you please update this script entirely in your next version so that it does not depend on an implicit PWD for everything. Every directory manipulation call in a script like this should be operating on absolute paths from a directory passed in. Then if you want it to run from the current directory, get the cwd() at the top and pass that in if a `--dir` is not specified.

Reply:
Sure , will address in next revision